docs/source/deprecation.md

```diff
@@ -1,14 +1,164 @@
-## Deprecation
+# Deprecation
+
+The deprecation process is here to make (your) life of developers easier.
+It's not possible to write perfect solution for everything and as the project
+is evolving, it happens that some functionality needs to be changed, replaced
+or dropped completely. Such situations are inevitable. To reduce negative
+impact on your code, we introduce the deprecation process described below.
+
+## What does it mean that something is deprecated?
+
+When you deprecate something, the only thing that changes is that the
+deprecated entity is marked in the code as deprecated which can have
+additional impact,likemessages produced on the user's terminal,
+in the report, ... But the rest of the functionality is the same as before,
+until the entity is removed completely.
+
+## What is the deprecation process for leapp
+
+In case some public public functions or models are to be removed for any
+reason, these need to be marked as deprecated before the removal (if possible).
+In our projects, the deprecation will be applied only for functions and models
+that have been introduced in an official release (IOW, a functionality that
+has been merged into the upstream, but has been removed before the release
+or was marked as experimental all the time, is going to be removed without
+the deprecation state). The time period during which the deprecated entity
+won't be removed is at least 6 months. That doesn't mean we will remove
```

Comment:
You are not deciding anything. The time period is specified. When the protection time period ends, we can remove the deprecated functionality. But we do not have to remove it just for the purpose of the removal right? When we are not working on anything related, we could keep it there as it is if we want. From our POV, even more we should focus when we deliver the functionality to customer. Have an example: we deprecate something on Jan. The release happens on April. Another release will happen on July. Does it means we have to or should remove the stuff for the July release? No. We can keep it for example for the next November release. So people have enough time to fix their code.

Reply:
@ZhukovGreen the date when you deprecate the functionality. e.g. the current day ;-) in case we would like to improve it, we can update the information before the release, in case it matters

Reply:
Hmm.. the date doesn't make sense. It's deprecated in the day of the commit. So since == today. not 6months in future.

Reply:
resolved over the call